### PR TITLE
nixos/boot: qemu-flags -> qemu-common

### PR DESCRIPTION
**Pull request by Artturin**
and dont use with import ...

###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/127933#issuecomment-930119297

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
hmm? 
`nix-build nixos/tests/boot.nix`
```> /nix/store/r28nip42zvcr35lsndbgafqa9w4d91n0-stdenv-linux/setup: line 88: makeWrapper: command not found```

**Review by dasJ — APPROVED** [review]

**Comment by dasJ** [comment]
The makeWrapper thing is from #135877

**Comment by Artturin** [comment]
works
```
/nix/store/wgxs8drcl5il5lpnm6r2k97cv0wfanjp-vm-test-run-boot-bios-cdrom
/nix/store/63zqjj6bw0n392flnnh7j9di95vgjdpf-vm-test-run-boot-netboot-bios
/nix/store/6f2qfzzqi4kgklx28vpysbcjjg3ssk62-vm-test-run-boot-bios-usb
/nix/store/i186v843q4pck6lfwqfxrkyxksjhlld2-vm-test-run-boot-uefi-cdrom
/nix/store/ppnhyajw1y1n95cwl4ycwswwnwpkiv9v-vm-test-run-boot-netboot-uefi
/nix/store/52mhcyy1r96lrf5g594daydx5isrj3kr-vm-test-run-boot-uefi-usb
```

File: nixos/tests/boot.nix [hunk 1]
```diff
@@ -4,10 +4,10 @@
 }:
 
 with import ../lib/testing-python.nix { inherit system pkgs; };
-with import ../lib/qemu-flags.nix { inherit pkgs; };
 with pkgs.lib;
 
 let
+  qemu-common = import ../lib/qemu-common.nix { inherit (pkgs) lib pkgs; };
```
Comment (SuperSandro2000) [inline nixos/tests/boot.nix:10]: Maybe we can use callPackage here without inherit?

Comment (Artturin) [reply nixos/tests/boot.nix:10]: i did what was done in this commit [`b29c2f9` (#127933)](https://github.com/NixOS/nixpkgs/pull/127933/commits/b29c2f97c37f7cb4a1b3411ff9888a49873597d2)

**Comment by Artturin** [comment]
Merging so that tests aren't broken

**Comment by dasJ** [comment]
Apart from one job that is still building, this seems to have fully unblocked unstable-small :)

**Pull request merged**